### PR TITLE
Fix codespaces build by bumping clang llvm

### PR DESCRIPTION
**Pull request by ilonatommy**
Command e.g. `./build.sh mono+libs -c release -os browser`

Without this change:
```
  In file included from /workspaces/runtime/artifacts/obj/mono/browser.wasm.Release/llvm//x64/include/c++/v1/stdint.h:106:
  /workspaces/runtime/artifacts/obj/mono/browser.wasm.Release/llvm//x64/include/c++/v1/__config:48:8: warning: "Libc++ only supports Clang 16 and later" [-W#warnings]
  #      warning "Libc++ only supports Clang 16 and later"
         ^
....
  /workspaces/runtime/artifacts/obj/mono/browser.wasm.Release/llvm//x64/include/c++/v1/new:227:14: error: 'std' is not a class, namespace, or enumeration
  operator new(std::size_t __sz, std::align_val_t) _THROW_BAD_ALLOC;
               ^
  /workspaces/runtime/artifacts/obj/mono/browser.wasm.Release/llvm//x64/include/c++/v1/__fwd/functional.h:18:1: note: 'std' declared here
  _LIBCPP_BEGIN_NAMESPACE_STD
  ^
  /workspaces/runtime/artifacts/obj/mono/browser.wasm.Release/llvm//x64/include/c++/v1/__config:853:81: note: expanded from macro '_LIBCPP_BEGIN_NAMESPACE_STD'
                                        namespace _LIBCPP_TYPE_VISIBILITY_DEFAULT std {                                  \
                                                                                  ^
  fatal error: too many errors emitted, stopping now [-ferror-limit=]
  1 warning and 20 errors generated.
  [333/385] Building C object mono/mini/CMakeFiles/monosgen-objects.dir/interp/transform-opt.c.o
  [334/385] Building C object mono/mini/CMakeFiles/monosgen-objects.dir/interp/interp.c.o
  [335/385] Building C object mono/mini/CMakeFiles/monosgen-objects.dir/interp/transform.c.o
  [336/385] Building C object mono/mini/CMakeFiles/monosgen-objects.dir/mini-llvm.c.o
  [337/385] Building C object mono/mini/CMakeFiles/monosgen-objects.dir/method-to-ir.c.o
  ninja: build stopped: subcommand failed.
/workspaces/runtime/src/mono/mono.proj(1049,5): error MSB3073: The command " cmake --build . --target install --config Release" exited with code 1.

Build FAILED.
```

With this change:
build succeeds.

Both are tested with applying the fix from https://github.com/dotnet/runtime/pull/105110.

Ubuntu 22.04 does not have system clang/llvm newer than 14.0, so we have to manually add repo and update packages from it. I took v=18 because that's what we have on CI.

### CODE REVIEW

**Comment by kg** [comment]
We should think about updating the documentation on how to build dotnet/runtime on linux at some point too. The current requirements just say to install clang, but it will be more complex now.

https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/linux-requirements.md

I don't think that needs to happen in this PR necessarily though.

File: .devcontainer/wasm-multiThreaded/Dockerfile [hunk 1]
```diff
@@ -5,12 +5,19 @@
 ARG VARIANT="6.0-jammy"
 FROM mcr.microsoft.com/devcontainers/dotnet:0-${VARIANT}
 
+# Set up machine requirements to build the repo and the gh CLI
+# Clang-16 up is rquired but Ubuntu 22.04 comes with clang-14 highest, so add clang-18 sources
```
Comment (radekdoulik) [inline .devcontainer/wasm-multiThreaded/Dockerfile:9]: nit: typo and the same below

**Review by radekdoulik — APPROVED** [review]

**Comment by akoeplinger** [comment]
Note that the clang-16 requirement only applies to builds where we use our llvm, i.e. wasm and mobile. On Desktop platforms an earlier clang should be fine still.

**Comment by ilonatommy** [comment]
> Note that the clang-16 requirement only applies to builds where we use our llvm, i.e. wasm and mobile. On Desktop platforms an earlier clang should be fine still.

You mean that the doc is misleading now? Running codespaces with newer clang than 14 for desktop won't do any harm I think. We don't have a better, wasm-only document. DO you think we should update
```
clang 16 or newer
```
to
```
clang (for WASM 16 or newer)
```
?

**Comment by akoeplinger** [comment]
Yeah I was mostly concerned about the docs

**Pull request merged**